activesupport/lib/active_support/cache.rb

```diff
@@ -478,7 +478,14 @@ def fetch(name, options = nil, &block)
           if entry
             get_entry_value(entry, name, options)
           else
-            save_block_result_to_cache(name, options, &block)
+            options = options.dup
+
+            result = instrument(:generate, key, options) do
+              yield(name, WriteOptions.new(options))
+            end
+
+            write(name, result, options) unless result.nil? && options[:skip_nil]
+            result
```

Comment:
I inlined this method so that `key` could be passed to `instrument`, but I wasn't sure if the method was something that should be kept.

Reply:
The `fetch` method is already rather long.  My inclination would be to keep `save_block_result_to_cache` and either add an extra param or just recompute `key = normalize_key(name, options)` inside it (depending on the overhead).

The subsequent `write` will recompute the normalized key too.  I wonder if there's some elegant way to avoid that?  :thinking:

Reply:
Yeah I had the same thought, maybe we should extract a private `_write` that both `fetch` and `write` can call with an already normalized key and merged options? Anyways, I'll put the method back with an extra parameter in the meantime.